doc/releases/release-notes-2.1.rst

```diff
@@ -17,53 +17,304 @@ Kernel
 ******
 
 * TBD
+19807 : kernel: activate arch interface headers : andrewboie : Andrew Boie
```

Comment:
Need a blank line after the last item in a bullet list.

Reply:
These were meant to be just highlight notes to the community to remind them of PRs that appear to belong to this heading. I didn't attempt to format them like we do with the issues closed section.

I'll clarify on next update